MockPyth.sol

```diff
@@ -7,21 +7,50 @@ import "./PythStructs.sol";
 contract MockPyth is AbstractPyth {
     mapping(bytes32 => PythStructs.PriceFeed) priceFeeds;
 
+    uint singleUpdateFeeInWei;
+    uint singleUpdateFeeInPythToken;
+
+    constructor(uint _singleUpdateFeeInWei, uint _singleUpdateFeeInPythToken) {
+        singleUpdateFeeInWei = _singleUpdateFeeInWei;
+        singleUpdateFeeInPythToken = _singleUpdateFeeInPythToken;
+    }
+
     function queryPriceFeed(bytes32 id) public override view returns (PythStructs.PriceFeed memory priceFeed) {
         return priceFeeds[id];
     }
 
     // Takes an array of encoded price feeds and stores them.
     // You can create this data either by calling createPriceFeedData or
     // by using web3.js or ethers abi utilities.
-    function updatePriceFeeds(bytes[] memory updateData) public override {
+    function updatePriceFeeds(bytes[] memory updateData, uint feeAmount, bool feeUsingPythToken) public override payable {
+        require(feeAmount >= getMinUpdateFee(updateData.length, feeUsingPythToken), "Insufficient fee amount");
+
+        if (feeUsingPythToken) {
+            // It doesn't do the actual transfer here, to avoid extra dependency.
+            // SafeERC20.safeTransferFrom(IERC20(pythToken), msg.sender, address(this), feeAmount);
+        } else {
+            require(msg.value >= feeAmount , "Fee amount is bigger than the payed wei");
+
+            // Paying back the remaining amount to the sender.
+            uint rem = msg.value - feeAmount;
+            payable(msg.sender).transfer(rem);
```

Comment:
this confuses me. Isn't `msg.value` redundant with `feeAmount` here?

Reply:
feeAmount here is redundant and I used it as a sanity check. I saw wormhole does it so did I. If we don't have token here then it's not needed at all.